subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -2029,7 +2029,8 @@ static int handle_request(struct zoap_packet *request,
 	obj = get_engine_obj(path.obj_id);
 	if (!obj) {
 		/* No matching object found - ignore request */
-		return -ENOENT;
+		r = -ENOENT;
+		goto error;
```

Comment:
This patch could be merged to the first one:
"refactor handle_request() error handling "

Reply:
Done